lib/db.js

```diff
@@ -479,19 +482,50 @@ module.exports = (
       uaDeviceType: token.uaDeviceType,
       lastAccessTime: token.lastAccessTime,
       createdAt: token.createdAt
-    }]
+    }
     let sessionTokens = []
+    let matchedTokens = []
+    const ipHash = ip && crypto.createHash('sha256').update(ip).digest('base64')
```

Comment:
If we want to store this, we'll have to get data-steward signoff.  FWIW we have an established pattern for "anonymizing" ip addreses here: https://github.com/mozilla/fxa-auth-db-mysql/blob/master/lib/db/mysql.js#L1145

I mentioned to @vladikoff though, that I would expect doing the ip -> location lookup to be quite fast and maybe we can afford to do it every time rather than keying a cache off the IP address at all.